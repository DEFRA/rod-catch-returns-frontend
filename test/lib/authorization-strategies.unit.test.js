describe('authorization-strategies', () => {
  const OLD_ENV = process.env

  beforeEach(() => {
    jest.resetModules()
    jest.clearAllMocks()
    process.env = { ...OLD_ENV } // make a copy
  })

  afterEach(() => {
    process.env = OLD_ENV // restore old env
  })

  describe('sessionCookie', () => {
    it('should have the correct config', () => {
      const authorizationStrategies = require('../../src/lib/authorization-strategies')
      expect(authorizationStrategies.sessionCookie).toMatchSnapshot()
    })

    it('should set cookie password if COOKIE_PW environment variable is present', () => {
      process.env.COOKIE_PW = 'cookie_password'
      const authorizationStrategies = require('../../src/lib/authorization-strategies')
      expect(authorizationStrategies.sessionCookie.cookie.password).toBe('cookie_password')
    })

    it('should not set cookie password if COOKIE_PW environment variable is not present', () => {
      delete process.env.COOKIE_PW
      const authorizationStrategies = require('../../src/lib/authorization-strategies')
      expect(authorizationStrategies.sessionCookie.cookie.password).toBeUndefined()
    })

    it('should set cookie isSecure to true if HTTPS environment variable is true', () => {
      process.env.HTTPS = 'true'
      const authorizationStrategies = require('../../src/lib/authorization-strategies')
      expect(authorizationStrategies.sessionCookie.cookie.isSecure).toBeTruthy()
    })

    it('should set cookie isSecure to false if HTTPS environment variable is not set', () => {
      delete process.env.HTTPS
      const authorizationStrategies = require('../../src/lib/authorization-strategies')
      expect(authorizationStrategies.sessionCookie.cookie.isSecure).toBeFalsy()
    })
  })

  describe('adminCookie', () => {
    it('should have the correct config', () => {
      const authorizationStrategies = require('../../src/lib/authorization-strategies')
      expect(authorizationStrategies.adminCookie).toMatchSnapshot()
    })

    it('should set cookie password if COOKIE_PW environment variable is present', () => {
      process.env.COOKIE_PW = 'cookie_password'
      const authorizationStrategies = require('../../src/lib/authorization-strategies')
      expect(authorizationStrategies.adminCookie.cookie.password).toBe('cookie_password')
    })

    it('should not set cookie password if COOKIE_PW environment variable is not present', () => {
      delete process.env.COOKIE_PW
      const authorizationStrategies = require('../../src/lib/authorization-strategies')
      expect(authorizationStrategies.adminCookie.cookie.password).toBeUndefined()
    })

    it('should set cookie isSecure to true if HTTPS environment variable is true', () => {
      process.env.HTTPS = 'true'
      const authorizationStrategies = require('../../src/lib/authorization-strategies')
      expect(authorizationStrategies.adminCookie.cookie.isSecure).toBeTruthy()
    })

    it('should set cookie isSecure to false if HTTPS environment variable is not set', () => {
      delete process.env.HTTPS
      const authorizationStrategies = require('../../src/lib/authorization-strategies')
      expect(authorizationStrategies.adminCookie.cookie.isSecure).toBeFalsy()
    })

    it('should set cookie isHttpOnly to true if HTTPS environment variable is true', () => {
      process.env.HTTPS = 'true'
      const authorizationStrategies = require('../../src/lib/authorization-strategies')
      expect(authorizationStrategies.adminCookie.cookie.isHttpOnly).toBeTruthy()
    })

    it('should set cookie isHttpOnly to false if HTTPS environment variable is not set', () => {
      delete process.env.HTTPS
      const authorizationStrategies = require('../../src/lib/authorization-strategies')
      expect(authorizationStrategies.adminCookie.cookie.isHttpOnly).toBeFalsy()
    })
  })
})
