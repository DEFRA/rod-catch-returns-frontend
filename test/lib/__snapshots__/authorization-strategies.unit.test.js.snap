// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authorization-strategies adminCookie should have the correct config 1`] = `
Object {
  "appendNext": true,
  "cookie": Object {
    "isHttpOnly": false,
    "isSameSite": "Lax",
    "isSecure": false,
    "name": "sid",
    "password": "cookieabcdef1234",
    "path": "/",
    "ttl": null,
  },
  "redirectTo": "/login",
}
`;

exports[`authorization-strategies sessionCookie should have the correct config 1`] = `
Object {
  "cookie": Object {
    "clearInvalid": true,
    "isSecure": false,
    "name": "sid",
    "password": "cookieabcdef1234",
  },
  "redirectTo": "/licence-auth",
  "validateFunc": [Function],
}
`;
